### PR TITLE
Migrate Linux and windows Docker build jobs from CircleCi to GitHub Action.

### PR DESCRIPTION
**Pull request by nitaliya**
Migrate Linux and windows Docker build jobs
Ticket: [OTL-1259](https://signalfuse.atlassian.net/browse/OTL-1259).

Adding GitHub Action jobs of Linux and windows Docker build
Removing CircleCi jobs of Linux and windows Docker build.

### CODE REVIEW

**Comment by jcheng-splunk** [comment]
@nitaliya Please rebase and delete the circleci config if there are all jobs have been migrated.

**Comment by nitaliya** [comment]
> @nitaliya Please rebase and delete the CircleCI config-if all jobs have been migrated.

Hello @jcheng-splunk,

Thanks for the heads up and yes we rebase the PR as a part of the routine steps that we follow on our end. For the multiple PRs that we raise in the same repository when one of those PRs gets approved and merged from your end, all the other open PRs get out of sync and will show behind the PR that got merged.

For this PR, I rebased it this morning. and As all jobs are migrated I have removed the config file of CircleCI. If there are no further changes needed, please approve and merge this PR.

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -433,3 +433,115 @@ jobs:
...
+          [ "$( docker inspect --format='{{.Architecture}}' otelcol:${{ matrix.ARCH }} )" = "${{ matrix.ARCH }}" ] || exit 1
+          docker run --platform linux/${{ matrix.ARCH }} -d -e SPLUNK_ACCESS_TOKEN=12345 -e SPLUNK_REALM=fake-realm --name otelcol otelcol:${{ matrix.ARCH }}
+          sleep 10
+          if [ -z "$( docker ps --filter=status=running --filter=name=otelcol -q )" ]; then
+            docker logs otelcol
+            echo "Failing job execution: fail to start otelcol docker container in 10 seconds."
+            exit 1
+          fi
+      
+      - name: Save docker image
```
Comment (jcheng-splunk) [inline .github/workflows/build-and-test.yml:490]: Let's skip saving and archiving the image to save time.

File: .github/workflows/build-and-test.yml [hunk 2]
```diff
@@ -433,3 +433,115 @@ jobs:
...
+          Start-Sleep 10
+          $DockerOutput=$(docker ps --filter=status=running --filter=name=otelcol -q)
+          if ( $DockerOutput -eq $null )
+          {
+            docker logs otelcol
+            echo "Failing job execution: fail to start otelcol docker container in 10 seconds."
+            exit 1
+          }
+      
+      - name: Save docker image
```
Comment (jcheng-splunk) [inline .github/workflows/build-and-test.yml:537]: Let's skip saving and archiving the image to save time.

File: .github/workflows/build-and-test.yml (hunk 1 above)
Comment (nitaliya) [reply .github/workflows/build-and-test.yml]: Removed save docker image step for the GitHub Action workflow now.

File: .github/workflows/build-and-test.yml (hunk 2 above)
Comment (nitaliya) [reply .github/workflows/build-and-test.yml]: Removed save docker image step for the GitHub Action workflow now.

**Review by jcheng-splunk — APPROVED** [review]
Thanks @nitaliya. We'll need to coordinate to disable the circleci checks and builds before merging.

**Pull request merged**